Domain/Sources/Domain/Protocols/NibLoadableView.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  NibLoadableView.swift
+//
+//  Created by heba isaa on 07/09/2021.
+//
+
+import UIKit
+public  protocol NibLoadableView: class {
+    static var nibName: String { get }
+}
+
+public extension NibLoadableView where Self: UIView {
```

Comment:
As you should know, `Protocol` with default implementation in `Extension` is one of most powerful features of Swift which enabled POP, but. Do you know what are the downsides of POP? (It's a very good topic to look for). :)

Reply:
You can't imagine how happy I am to see your valuable feedback here.. Thanks ya Karim ❤️ 
@KarimEbrahemAbdelaziz

Reply:
Thank you ,I will check it 

> As you should know, `Protocol` with default implementation in `Extension` is one of most powerful features of Swift which enabled POP, but. Do you know what are the downsides of POP? (It's a very good topic to look for). :)

great
thank you ,I will check it